did/ion/operations.go

```diff
@@ -0,0 +1,324 @@
+package ion
+
+import (
+	"bytes"
+	"fmt"
+	"io"
+	"net/http"
+	"net/url"
+	"reflect"
+	"strings"
+
+	"github.com/TBD54566975/ssi-sdk/crypto"
+	"github.com/TBD54566975/ssi-sdk/did"
+	"github.com/TBD54566975/ssi-sdk/util"
+	"github.com/goccy/go-json"
+	"github.com/pkg/errors"
+)
+
+type (
+	DIDION string
+)
+
+const (
+	IONPrefix = "did:ion"
+)
+
+// IsValid checks if the did:ion is valid by checking for a valid prefix
+// full validation is impossible without resolution
+func (d DIDION) IsValid() bool {
+	split := strings.Split(d.String(), IONPrefix+":")
+	return len(split) == 2
+}
+
+func (d DIDION) String() string {
+	return string(d)
+}
+
+func (d DIDION) Suffix() (string, error) {
+	split := strings.Split(d.String(), IONPrefix+":")
+	if len(split) != 2 {
+		return "", errors.Wrap(util.InvalidFormatError, "did is malformed")
+	}
+	return split[1], nil
+}
+
+func (DIDION) Method() did.Method {
+	return did.IONMethod
+}
+
+type Resolver struct {
+	baseURL string
+}
+
+// NewIONResolver creates a new resolver for the ION DID method with a common base URL
+// The base URL is the URL of the ION node, for example: https://ion.tbd.network
+// The resolver will append the DID to the base URL to resolve the DID such as
+//
+//	https://ion.tbd.network/identifiers/did:ion:1234
+//
+// and similarly for submitting anchor operations to the ION node...
+//
+//	https://ion.tbd.network/operations
+func NewIONResolver(baseURL string) (*Resolver, error) {
+	if _, err := url.ParseRequestURI(baseURL); err != nil {
+		return nil, errors.Wrap(err, "invalid resolver URL")
+	}
+	return &Resolver{baseURL: baseURL}, nil
+}
+
+// Resolve resolves a did:ion DID by appending the DID to the base URL with the identifiers path and making a GET request
+func (i Resolver) Resolve(id string, _ did.ResolutionOptions) (*did.DIDResolutionResult, error) {
+	if i.baseURL == "" {
+		return nil, errors.New("resolver URL is empty")
+	}
+	resp, err := http.Get(strings.Join([]string{i.baseURL, "identifiers", id}, "/"))
+	if err != nil {
+		return nil, errors.Wrapf(err, "could not resolve with URL: %s", i.baseURL)
+	}
+
+	defer resp.Body.Close()
+	body, err := io.ReadAll(resp.Body)
+	if err != nil {
+		return nil, errors.Wrapf(err, "could not resolve with response %+v", resp)
+	}
+	if !is2xxStatusCode(resp.StatusCode) {
+		return nil, fmt.Errorf("could not resolve DID: %s", string(body))
+	}
+	resolutionResult, err := did.ParseDIDResolution(body)
+	if err != nil {
+		return nil, errors.Wrapf(err, "resolving did:ion DID<%s>", id)
+	}
+	return resolutionResult, nil
+}
+
+// Anchor submits an anchor operation to the ION node by appending the operations path to the base URL
+// and making a POST request
+func (i Resolver) Anchor(op AnchorOperation) error {
```

Comment:
A top level package example on how to use this package would be awesome. Something super simple like: here's how you create an ION did: 

```
NewIONDID()
Anchor()
```